### PR TITLE
Bugfix FXIOS-4383 [v117] Pinch to zoom out can cause a page reload

### PR DESCRIPTION
**Pull request by aalaaeid**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-4383)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/11007)

## :bulb: Description
This fixes reloading the page when pinch zooming out by disabling the refresh control during pinch zooming, and enabling it for swipe gestures only.

https://github.com/mozilla-mobile/firefox-ios/assets/37640295/745f0482-8457-45b8-8d96-59dd1bb7f4d0

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Browser/TabScrollController.swift
Comment (thatswinnie) [inline Client/Frontend/Browser/TabScrollController.swift:415]: We should take the feature flag into account when re-enabling the refresh control.

**Review by thatswinnie — APPROVED** [review]

**Comment by lmarceau** [comment]
Added "Do not merge" until we have UX approval to merge the PR. This looks great @aalaaeid 🙌

**Comment by lmarceau** [comment]
Those changes will be accessible through TestFlight Nightly #32972

**Review by topotropic — APPROVED** [review]
I tested it via TestFlight and didn't run into any issues, looks like we decoupled pinch-to-zoom and pull-to-refresh so that they don't interfere, thanks for that!

**Pull request merged**
